### PR TITLE
Update WalletConnect to v2

### PR DESCRIPTION
**Pull request by michalsmiarowski**
<p align="center">
&#x1f537; WalletConnect V2 implementation &#x1f537;
</p>

Hero Horizontal

This pull request updates our current Wallet Connect implementation to version 2, incorporating the latest features and improvements. It ensures compatibility with the latest standards and provides a better user experience for our wallet integration.

### Changes Made
- Created a new connector - `WalletConnectConnector` - that is compatible with `web3-react` version 6 which we use
- Added new libraries - `@walletconnect/ethereum-provider` and `@walletconnect/modal` - which helped us implement the new version
- Removed everything related to WalletConnect v1 (this included `@web3-react/walletconnect-connector` lib)
- Added new .env variable `WALLET_CONNECT_PROJECT_ID` that will store our project id needed for WalletConnect v2

### CODE REVIEW

File: src/web3/connectors/walletConnect.ts [hunk 1]
```diff
@@ -1,18 +1,194 @@
...
+    this.handleOnDisplayUri = this.handleOnDisplayUri.bind(this)
+
+    this.handleChainChanged = this.handleChainChanged.bind(this)
+    this.handleAccountsChanged = this.handleAccountsChanged.bind(this)
+    this.handleDisconnect = this.handleDisconnect.bind(this)
+  }
+
+  private handleOnConnect(): void {}
+
+  private handleOnDisplayUri(): void {}
```
Comment (r-czajkowski) [inline src/web3/connectors/walletConnect.ts:61]: Maybe we should emit an event here?

File: src/web3/connectors/walletConnect.ts [hunk 2]
```diff
@@ -1,18 +1,194 @@
...
+    const account = await new Promise<string>((resolve, reject) => {
+      const userReject = () => {
+        // Erase the provider manually
+        this.provider = undefined
+        reject(new UserRejectedRequestError())
+      }
+
+      // Workaround to bubble up the error when user reject the connection
+      this.provider!.on("disconnect", () => {
+        // Check provider has not been enabled to prevent this event callback from being called in the future
```
Comment (r-czajkowski) [inline src/web3/connectors/walletConnect.ts:114]: Let's break this long comment.

**Review by r-czajkowski — CHANGES_REQUESTED** [review]
Lef minor comments. Overall LGTM.

File: src/web3/connectors/walletConnect.ts (hunk 2 above)
Comment (michalsmiarowski) [reply src/web3/connectors/walletConnect.ts]: 3525fb966c590b5f337728dfede61a3a10948bfb

File: package.json [hunk 3]
```diff
@@ -43,12 +43,14 @@
     "@types/react": "^17.0.20",
     "@types/react-dom": "^17.0.9",
     "@types/react-router-dom": "^5.3.2",
+    "@walletconnect/ethereum-provider": "^2.8.6",
+    "@walletconnect/modal": "^2.5.9",
+    "@walletconnect/types": "1.7.8",
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply package.json]: I removed the `@walletconnect/types` completely in 2aff7383b9e3f9807409e60c8af9421181fea0fe

File: src/web3/connectors/walletConnect.ts [hunk 4]
```diff
@@ -1,18 +1,194 @@
...
+    this.provider.on("connect", this.handleOnConnect)
+    this.provider.on("display_uri", this.handleOnDisplayUri)
+
+    // ensure that the uri is going to be available, and emit an event if there's a new uri
+    // if (!this.walletConnectProvider.connected) {
+    //   await this.walletConnectProvider.connector.createSession(
+    //     this.config.chainId ? { chainId: this.config.chainId } : undefined
+    //   )
+    //   this.emit(URI_AVAILABLE, this.walletConnectProvider.connector.uri)
+    // }
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply src/web3/connectors/walletConnect.ts]: e7edb2e2941baf2651261e354cfaecd5afda118e

**Review by r-czajkowski — APPROVED** [review]

File: src/web3/connectors/walletConnect.ts (hunk 1 above)
Comment (michalsmiarowski) [reply src/web3/connectors/walletConnect.ts:57]: I will add those in a future PR when we add succes and error modals to the WalletConnec

**Review by pdyraga — APPROVED** [review]
Checked only workflow changes, and they all look good.

File: .github/workflows/reusable-build-and-publish.yml [hunk 5]
```diff
@@ -178,6 +183,7 @@ jobs:
           ELECTRUM_PORT: ${{ secrets.electrumPort }}
           SENTRY_SUPPORT: true
           SENTRY_DSN: ${{ secrets.sentryDsn }}
+          WALLET_CONNECT_PROJECT_ID: ${{ secrets.WALLET_CONNECT_PROJECT_ID }}
```
Comment (r-czajkowski) [inline .github/workflows/reusable-build-and-publish.yml:186]: Should be `${{ secrets.walletConnectProjectId }}`

File: .github/workflows/reusable-build-and-publish.yml [hunk 6]
```diff
@@ -194,6 +200,7 @@ jobs:
...
```
Comment (r-czajkowski) [inline .github/workflows/reusable-build-and-publish.yml:203]: Same as above.

**Review by r-czajkowski — CHANGES_REQUESTED** [review]

File: .github/workflows/reusable-build-and-publish.yml (hunk 6 above)
Comment (michalsmiarowski) [reply .github/workflows/reusable-build-and-publish.yml]: f49d3c666260ddaa3f624a470d21dfcc51bad98c

**Review by r-czajkowski — APPROVED** [review]

**Pull request merged**